frontend/pedigree.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { Component, useRef, useEffect } from 'react'
+import { Network } from 'vis-network';
+import "vis-network/styles/vis-network.css"
+import { useHistory } from "react-router-dom";
+
+
+const PedigreeNetwork = ({pedigree}) => {
```

Comment:
I don't know how much "typing coverage" you're aiming for in this project but adding types here can be done like so:

```typescript
const PedigreeNetwork = ({pedigree}: {pedigree: string}) => {...
```

Also if you define components with `function` you see that name in debugging views (profiler, componentsv iew and error messages).